modules/node_groups/templates/userdata.sh.tpl

```diff
@@ -1,9 +1,20 @@
 #!/bin/bash -e
+%{ if length(ami_id) == 0 || ami_is_eks_optimized ~}
 
-# Allow user supplied pre userdata code
+sed -i '/^KUBELET_EXTRA_ARGS=/a KUBELET_EXTRA_ARGS+=" ${kubelet_extra_args}"' /etc/eks/bootstrap.sh
+%{endif ~}
+%{if length(pre_userdata) > 0 ~}
```

Comment:
I don't think we need to have this `if/endif` block but just include `${pre_userdata}` always as it was before.

Reply:
It would work without the if block, but it's tidier to use it so the resulting script has consistent white space. It also makes it clear in the template that it's an optional variable.